processing/src/main/java/io/druid/segment/filter/DimensionPredicateFilter.java

```diff
@@ -37,14 +38,17 @@
 {
   private final String dimension;
   private final Predicate<String> predicate;
+  private final ExtractionFn extractionFn;
 
   public DimensionPredicateFilter(
       String dimension,
-      Predicate<String> predicate
+      Predicate<String> predicate,
+      ExtractionFn extractionFn
```

Comment:
cause extraction itself is handled in predicate, I think we don't need this here (can be confusing).